sim/midas/src/main/scala/midas/models/dram/Util.scala

```diff
@@ -752,3 +819,59 @@ class MemoryModelMonitor(cfg: BaseConfig)(implicit p: Parameters) extends MultiI
   assert(!axi4.aw.fire || axi4.aw.bits.len < cfg.maxWriteLength.U,
     s"Write burst length exceeds memory-model maximum of ${cfg.maxReadLength}")
 }
+
+class FIFOAlignedAddressMatcherUnitTest(val tranXCount : Int = 2000)(implicit p: Parameters) extends UnitTest {
+
+  val addrBits = 64
+  val sizeBits = 2
+  val lenBits = 8
+  val nastiP = p.alterPartial({
+    case NastiKey => NastiParameters(64, 16, 4)
+  })
+
+  val tranXs = RegInit(0.U(12.W))
+
+  val enqdeq = RegInit(false.B)
```

Comment:
Ideally you'd fuzz enq and deq times too.

Reply:
Done